src/main/java/gregtech/api/unification/material/materials/SecondDegreeMaterials.java

```diff
@@ -0,0 +1,400 @@
+import gregtech.api.unification.material.properties.PropertyKey;
+
+import static gregtech.api.unification.material.Materials.*;
+import static gregtech.api.unification.material.info.MaterialFlags.*;
+import static gregtech.api.unification.material.info.MaterialIconSet.*;
+
+public class SecondDegreeMaterials {
+
+    public static void register() {
+
+        Glass = new Material.Builder(2000, "glass")
+                .gem(0).fluid()
+                .color(0xFAFAFA).iconSet(GLASS)
+                .flags(GENERATE_LENS, NO_SMASHING, EXCLUDE_BLOCK_CRAFTING_RECIPES, DECOMPOSITION_BY_CENTRIFUGING)
+                .components(SiliconDioxide, 1)
+                .build();
+
+        Perlite = new Material.Builder(2001, "perlite")
+                .dust(1)
+                .color(0x1E141E)
+                .components(Obsidian, 2, Water, 1)
+                .build();
+
+        Borax = new Material.Builder(2002, "borax")
+                .dust(1)
+                .color(0xFAFAFA).iconSet(FINE)
+                .components(Sodium, 2, Boron, 4, Water, 10, Oxygen, 7)
+                .build();
+
+        Lignite = new Material.Builder(2003, "lignite")
+                .gem(0, 1200).ore() // 2/3 of burn time of coal
+                .color(0x644646).iconSet(LIGNITE)
+                .flags(FLAMMABLE, NO_SMELTING, NO_SMASHING, MORTAR_GRINDABLE)
+                .components(Carbon, 3, Water, 1)
+                .build();
+
+        Olivine = new Material.Builder(2004, "olivine")
+                .gem().ore(2, 1)
+                .color(0x96FF96).iconSet(RUBY)
+                .flags(NO_SMASHING, NO_SMELTING, HIGH_SIFTER_OUTPUT)
+                .components(Magnesium, 2, Iron, 1, SiliconDioxide, 2)
+                .toolStats(7.5f, 3.0f, 312)
+                .build();
+
+        Opal = new Material.Builder(2005, "opal")
+                .gem().ore()
+                .color(0x0000FF).iconSet(OPAL)
+                .flags(NO_SMASHING, NO_SMELTING, HIGH_SIFTER_OUTPUT, DECOMPOSITION_BY_CENTRIFUGING)
+                .components(SiliconDioxide, 1)
+                .toolStats(7.5f, 3.0f, 312)
+                .build();
+
+        Amethyst = new Material.Builder(2006, "amethyst")
+                .gem(3).ore()
+                .color(0xD232D2).iconSet(RUBY)
+                .flags(NO_SMASHING, NO_SMELTING, HIGH_SIFTER_OUTPUT)
+                .components(SiliconDioxide, 4, Iron, 1)
+                .toolStats(7.5f, 3.0f, 312)
+                .build();
+
+        Lapis = new Material.Builder(2007, "lapis")
+                .gem(1).ore(6, 4)
+                .color(0x4646DC).iconSet(LAPIS)
+                .flags(NO_SMASHING, NO_SMELTING, CRYSTALLIZABLE, NO_WORKING, DECOMPOSITION_BY_ELECTROLYZING, EXCLUDE_BLOCK_CRAFTING_BY_HAND_RECIPES)
+                .components(Lazurite, 12, Sodalite, 2, Pyrite, 1, Calcite, 1)
+                .build();
+
+        Blaze = new Material.Builder(2008, "blaze")
+                .dust(1).fluid()
```

Comment:
Since it already has some extra work put into being a fluid and some uses, I will leave it for now, but this is something to consider in the future for sure